avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -452,15 +464,28 @@ private static Object fromConnectData(Schema schema, org.apache.avro.Schema avro
           Struct struct = (Struct) value;
           if (!struct.schema().equals(schema))
             throw new DataException("Mismatching struct schema");
-          org.apache.avro.Schema underlyingAvroSchema = avroSchemaForUnderlyingTypeIfOptional(schema, avroSchema);
-          GenericRecordBuilder convertedBuilder = new GenericRecordBuilder(underlyingAvroSchema);
-          for (Field field : schema.fields()) {
-            org.apache.avro.Schema fieldAvroSchema = underlyingAvroSchema.getField(field.name()).schema();
-            convertedBuilder.set(
-                field.name(),
-                fromConnectData(field.schema(), fieldAvroSchema, struct.get(field), false, true));
+          if (AVRO_TYPE_UNION.equals(schema.name())) {
+            Object object;
+            for (Field field : schema.fields())
+            {
```

Comment:
Do you have an intellij code style config file to share? Then i can simply reformat in intellij using that :)